scrapyjs/middleware.py

```diff
@@ -43,24 +52,26 @@ def from_crawler(cls, crawler):
         return cls(crawler, splash_base_url, slot_policy)
 
     def process_request(self, request, spider):
-        splash_options = request.meta.get('splash')
+        splash_options = self.get_splash_options(request, spider)
         if not splash_options:
             return
 
+        elif request.meta.get("_splash_processed"):
+            return
+
         if request.method != 'GET':
             log.msg("Currently only GET requests are supported by SplashMiddleware; %s "
                     "will be handled without Splash" % request, logging.WARNING)
             return request
 
         meta = request.meta
-        del meta['splash']
-        meta['_splash_processed'] = splash_options
```

Comment:
you added check `if request.meta.get("_splash_processed"): return` above - isn't that enough. 

1. request processed first time, 'splash' moved to '_splash_processed'
2. response processed first time - treat '_splash_processed' as dict of splash options
3. request processed second time (any reason like retry) - '_splash_processed' is already in meta so request will be ignored because of the check `request.meta.get("_splash_processed")`

but after your explanation I think the way it's implemented in PR is better, because someone might expect to have 'splash' key in response.meta, and with current implementation it's missed. @kmike are you okay with that?